torrent.go

```diff
@@ -1252,6 +1263,11 @@ func (t *Torrent) announceDHT(impliedPort bool) {
 					allAddrs[key] = struct{}{}
 				}
 				cl.mu.Lock()
+				if t.peers == nil { // torrent can be already closed. return.
```

Comment:
I don't think you should use zeroed state to indicate something being closed. Just test Torrent.closed directly if it matters. Here, Torrent.addPeers should just not trigger methods like openNewConns etc. if it's closed.